Data/ByteString.hs

```diff
@@ -442,17 +442,18 @@ append = mappend
 -- | /O(n)/ 'map' @f xs@ is the ByteString obtained by applying @f@ to each
 -- element of @xs@.
 map :: (Word8 -> Word8) -> ByteString -> ByteString
-map f (BS fp len) = unsafeDupablePerformIO $ unsafeWithForeignPtr fp $ \a ->
-    create len $ map_ 0 a
+map f (BS fp len) = unsafeDupablePerformIO $ unsafeWithForeignPtr fp $ \ptr1 ->
+    create len $ \ptr2 -> m ptr1 ptr2
   where
-    map_ :: Int -> Ptr Word8 -> Ptr Word8 -> IO ()
-    map_ !n !p1 !p2
-       | n >= len = return ()
-       | otherwise = do
-            x <- peekByteOff p1 n
-            pokeByteOff p2 n (f x)
-            map_ (n+1) p1 p2
-{-# INLINE map #-}
+    m p1 p2 = map_ 0
```

Comment:
Please retain bangs, just in case.

Reply:
I can do that but I measured that it degrades performance.

Reply:
Could you please compare Core with and without bangs?

Reply:
I should note that the `Data.ByteString.Internal` function `findIndexOrEnd` doesn't have a bang pattern on the pointer argument so currently there is an inconsistent policy. Nevertheless I will add it.

Reply:
There could very well be discrepancies of such kind between functions, but we can hunt them down other day. For the purpose of current PR let's change as little as possible.